plugins/pluginmanagers/lz-n.nix

```
@@ -0,0 +1,106 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  lzPluginType =
+    with helpers.nixvimTypes;
+    submodule {
+      options = {
+        name = mkOption {
+          type = str;
+          description = "The plugin's name (not the module name). This is what is passed to the load(name) function.";
+        };
+        enabledInSpec = helpers.mkNullOrStrLuaFnOr bool ''
+          When false, or if the function returns false, then this plugin will not be included in the spec.
+          This option corresponds to the `enabled` property of lz.n.
+        '';
+        beforeAll = helpers.mkNullOrLuaFn "Always executed before any plugins are loaded.";
+        before = helpers.mkNullOrLuaFn "Executed before this plugin is loaded.";
+        after = helpers.mkNullOrLuaFn "Executed after this plugin is loaded.";
+        event = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+        cmd = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on command.";
+        ft = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on filetype.";
+        colorscheme = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on colorscheme.";
+        keys = helpers.mkNullOrOption (listOf helpers.keymaps.mapOptionSubmodule) "Lazy-load on key mapping. Use the same format as `config.keymaps`.";
+        priority = helpers.mkNullOrOption number ''
+          Only useful for start plugins (not lazy-loaded) to force loading certain plugins first. 
+          Default priority is 50 (or 1000 if colorscheme is set).
+        '';
```

```suggestion
        priority = helpers.defaultNullOpts.mkUnsignedInt (lib.literalMD "`50` (or `1000` if `colorscheme` is set)") ''
          Only useful for start plugins (not lazy-loaded) to force loading certain plugins first. 
        '';
```

We should try and use `defaultNullOpts` and provide a plugin default when defining options that directly map to upstream's options.

Even if the plugin default is unknown, you can use the `'` variant (e.g. `mkUnsignedInt'`) and not provide `pluginDefault`, though this is _effectively_ the same as using the non-`defaultNullOpts` option helpers.